### ISSUE TITLE
Ability to add specular maps to texture packs

### ISSUE THREAD

**Issue by tobyplowy** [minetest/minetest#3944]
Because I love bump maps but I really miss specular Maps

**Comment by C1ffisme** [comment]
Great. Have you got an example of these? Because most of us are just checking through our email/github messages, and don't want to open a browser just to search 'specular maps'.

**Comment by tobyplowy** [comment]
Here are a couple examples of specular Maps I hope it's useful https://www.google.be/search?q=specular+map+before+and+after&client=tablet-android-google&prmd=ivmn&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjDkpK-9O3LAhUM7xQKHTc5AIcQ_AUIBygB&biw=960&bih=600#tbm=isch&q=specular+map+textures&imgrc=ltfgavSIc2TwUM%3A 
  https://www.google.be/search?q=specular+map+before+and+after&client=tablet-android-google&prmd=ivmn&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjDkpK-9O3LAhUM7xQKHTc5AIcQ_AUIBygB&biw=960&bih=600#tbm=isch&q=specular+map+textures&imgrc=Q9YWO2BeSlV-5M%3A

**Comment by paramat** [comment]
This is only possible once we have a new lighting system with light sources, so a long way off.
Even then i consider it too specialised and fancy for Minetest, there would be many other effects to implement before specularity. Another issue is it forcing modders to create specular maps for every texture, this is why we removed designed bumpmaps from MTGame.

**Comment by tobyplowy** [comment]
Wait did you remove bump mapping No plz

**Comment by C1ffisme** [comment]
@tobyplowy Those links still require you to... open a browser.

I was talking maybe... screen shots from whatever game you use these in?

> Wait did you remove bump mapping No plz

We removed bump-mapping... Well... I'm not exactly sure why, but I imagine it was an under-developed shader that people disliked.

**Comment by 0-afflatus** [comment]
bump-maps, normal-maps and specularity-maps are lovely ideas. I'd love to have the kind of detailed textures that an engine like Cube can provide, but ... and it's a very big but, the minetest world is HUGE and part of the beauty of this thing is the brute-force rendering that means you can play a 3d game in a big world on relatively limited hardware. Also it has to be done well if we are going to do it and the 16x16px bump-maps looked really ugly to my eyes. Under the current system you can create a new node in about 10 minutes, so for example, I was recently able to upgrade all my doors in an afternoon. Having shiny bits and extra detail wouldn't improve the players' experience that much and the time saved allows me to concentrate on gameplay, which does hugely improve the game. 

So tl;dr please no.

**Comment by C1ffisme** [comment]
I agree with @0-afflatus. If I had to add a special map-map for every single node in a subgame, I would stop developing it. Immediately.

I don't have time to work on maps, I'd much rather prefer squashing bugs that I can accumulate, or balancing gameplay.

Besides that, since I have no knowledge or skill making these maps, I would have to look online for a tutorial (which would probably cost money, unless I can find something free) to learn how to map nodes.

So, yeah. And I know that this is optional, but we **do not want** to encourage players not to play on games without 'special textures'.

**Comment by paramat** [comment]
Bump mapping has not been removed, it just now uses automatically generated normalmaps, which of corse often look bad because they are auto-generated on pixel value.

**Comment by paramat** [comment]
No point keeping this open currently, please do open a new issue once we have a new lighting system, if that ever happens.
